frontend/src/scenes/trends/personsModalLogic.tsx

```diff
@@ -418,8 +453,31 @@ export const personsModalLogic = kea<personsModalLogicType<LoadPeopleFromUrlProp
                 saveOriginal,
                 searchTerm,
                 funnelStep,
+                crossDataset,
+                seriesId,
             })
         },
+        switchToDataPoint: async ({ seriesId }) => {
+            const data = values.people?.crossDataset?.find(({ seriesId: _id }) => _id === seriesId)
+
+            if (data && values.peopleParams) {
```

Comment:
If these conditions aren't met, we run the risk of showing stale persons in the modal once the data point has been switched. This can be confusing since users might believe they are looking at the new data point they just switched to. We should either return an empty list of people or prevent these data points from being shown in the dropdown in the first place.

Reply:
I believe these conditions should always be met, as we only show elements in the dropdown that come from `crossDataset`, so `data` should always be defined in theory.